packages/benchmarks/lit-html/kitchen-sink/index.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/**
+ * Kitchen-sink benchmark for lit-html
+ *
+ * Features exercised:
+ * - NodePart
+ *   - value: string
+ *   - value: Node
+ *   - value: Array<TemplateResult>
+ *   - value: directive: repeat
+ *   - value: nothing
+ * - AttributePart
+ *   - value: string (single)
+ *   - value: string (multiple)
+ *   - value: directive: classMap
+ *   - value: directive: styleMap
+ * - PropertyPart
+ *   - value: object
+ * - EventPart
+ *   - value: function
+ *
+ * Available query params:
+ * - width: number of items in each map/repeat per item
+ * - depth: number of levels of item recursion
+ * - updateCount: number of times to update after initial render
+ * - updateData: boolean, false to reuse same data each update, true to regenerate
+ */
+
+import {html, render, nothing} from 'lit-html';
+
+// TODO(kschaaf) use real repeat once landed
+// import { repeat } from "lit-html/lib/directives/repeat.js";
+const repeat = (items: any[], _kfn: (i: any) => any, tfn: (i: any) => any) =>
```

Comment:
Ah cool, haven't looked since I rebased. Lemme try using that.

Reply:
Well, I put it in but had to take it back out, because then the benchmark fails when run on tip-of-tree (which doesn't yet have it and so the import fails).

It raises a tricky issue with adding new features + benchmarks at the same time. As long as the features we're adding are all getting back to "feature parity", we can just add benchmarks in a following PR after the feature lands in tot.  But any actual new features can't really be put in the kitchen-sink benchmark, they'd need their own, so that we can configure it to not test against previous/tot branches.

Reply:
That makes sense and seems fine.

I don't think it's work worrying about now, but you could use dynamic import to get repeat if it's there. Maybe we use that if we need to benchmark a new feature in the future.

Reply:
Right I thought about that, but then the benchmark timing won't be comparable to when the feature doesn't exist.  Which is why kitchen-sink needs to be stable features.